web-ui/src/app/services/results.service.ts

```diff
@@ -278,7 +279,9 @@ type ApiSearchResult = [
     // 7 end pos iteration
     number,
     // 8 databases left to search
-    string[]
+    string[],
+    // 9 database of each hit
+    { [id: string]: string }
 ];
```

Comment:
I assume the data structure "results" is something that comes from old code base, so I'm not sure if it's possible: it would help readability and maintainability if it were clearer what the different elements in this array are.